packages/luna/src/components/Lessons/PastLessonsTable.tsx

```diff
@@ -10,51 +10,81 @@ import { isEmpty } from "lodash";
 import EmptyLessons from "@litespace/assets/EmptyLesson2";
 import { Link } from "react-router-dom";
 import { Loader, LoadingError } from "@/components/Loading";
-import { Void } from "@litespace/types";
+import { IUser, Void } from "@litespace/types";
 
 export type Props = {
   lessons: Array<{
     id: number;
     start: string;
     duration: number;
-    tutor: {
+    /**
+     * student or tutor or tutor manager
+     */
+    user: {
       id: number;
       name: string | null;
       imageUrl: string | null;
     };
+    peerId: number;
   }>;
-  onRebook?: (tutorId: number) => void;
   tutorsRoute: string;
+  /**
+   * role of user wheather tutor or tutor manager or student
+   */
+  userRole?: IUser.Role;
   loading?: boolean;
   error?: boolean;
+  /**
+   * state of sending message loading present in the end of each row of the table.
+   */
+  sendMessageLoading?: boolean;
+  /**
+   * state of sending message disable present in the end of each row of the table.
+   */
+  sendMessageDisable?: boolean;
```

Comment:
No need for this. We can derive the disabled state from the loading state in `sendingMessage`.